drivers/tty/serdev/pibridge.c

```diff
@@ -436,12 +437,26 @@ int pibridge_req_io(u8 addr, u8 cmd, u8 *snd_buf, u16 snd_len, u8 *rcv_buf,
 				"receive crc error in io-req\n");
 			return -EIO;
 		}
-		crc = pibridge_crc8(crc, rcv_buf, rcv_len);
-
-		if (crc != crc_rcv)
+		crc = pibridge_crc8(crc, rcv_buf, pkthdr.len);
+		if (crc != crc_rcv) {
+			dev_warn_ratelimited(&pibridge_s->serdev->dev,
+				"invalid checksum (expected: 0x%02x, got 0x%2x\n",
+				crc_rcv, crc);
 			return -EBADMSG;
+		}
+	}
+
+	if (pkthdr.addr != addr) {
+		dev_warn_ratelimited(&pibridge_s->serdev->dev,
+			"unexpeced response addr 0x%02x\n", pkthdr.addr);
+		return -EBADMSG;
+	}
+
+	if (!pkthdr.rsp) {
+		dev_warn_ratelimited(&pibridge_s->serdev->dev,
+			"no response flag set in received packet\n");
```

Comment:
Should probably be: "response flag *not* set in received packet"

Because "no response flag set" sounds like there could be multiple response flags but none of them is set.